### PR TITLE
JIT: Ensure correct alignment of CILJit singleton

### LINKED ISSUE #73409: placement new does not return properly aligned instance
### Description

https://github.com/dotnet/runtime/blob/9f1dd1aa499a882453b34d4e8810626423fbfed8/src/coreclr/jit/ee_il_dll.cpp#L163

Code in and around CILJitSingleton is returning a pointer in to a static char[] CILJitBuff, which in turn is returned from a placement new invocation.  operator new is supposed to return memory aligned to `__STDCPP_DEFAULT_NEW_ALIGNMENT__` which for linux x64 is 16.

When I link this code in CILJitBuff is not aligned.  However, my build system takes liberties, so perhaps there is an out-of-band linker specification somewhere that tells the linker to align to 0 mod 16.

For gcc/clang, putting
```
__attribute__ ((__aligned__(__STDCPP_DEFAULT_NEW_ALIGNMENT__)))
```
on the declaration of CILJitBuf makes the alignment problem go away, but this is unlikely to port to MSVC,

### Reproduction Steps

Compile with clang-14 -fsanitize=undefined and wait for the runtime checker to complain at the call to placement new.

### Expected behavior

should return properly aligned data, and no observed runtime error

### Actual behavior

run time error via -fsanitize=undefined and not suppressing all unaligned faults.

### Regression?

_No response_

### Known Workarounds

_No response_

### Configuration

_No response_

### Other information

/cc @aaronrob

### PR DESCRIPTION
**Pull request by jakobbotsch**
Fix #73409

### CODE REVIEW

**Comment by jkotas** [comment]
Linux arm failure is #73247

**Review by jkotas — APPROVED** [review]

File: src/coreclr/jit/ee_il_dll.cpp [hunk 1]
```diff
@@ -173,7 +161,8 @@ DLLEXPORT ICorJitCompiler* getJit()
 
     if (ILJitter == nullptr)
     {
-        ILJitter = new (CILJitSingleton) CILJit();
+        alignas(alignof(CILJit)) static char CILJitBuff[sizeof(CILJit)];
```
Comment (jkotas) [inline src/coreclr/jit/ee_il_dll.cpp:164]: Actually, is there a reason why we cannot just use a regular `static CILJit` singleton and just return its address here?

Comment (jkotas) [reply src/coreclr/jit/ee_il_dll.cpp:164]: https://github.com/dotnet/runtime/pull/73471

Comment (jakobbotsch) [reply src/coreclr/jit/ee_il_dll.cpp:164]: I was not sure if there were concerns with the potential static constructor or ordering with `jitStartup` (there is a check that initialization has run before we construct this).

Comment (jkotas) [reply src/coreclr/jit/ee_il_dll.cpp:164]: `CILJit` constructor does close to nothing, it is just initializing the vtable. So the ordering is not a concern.

**Pull request merged**